src/filesystem.coffee

```diff
@@ -28,23 +29,47 @@ class Filesystem
     node.unpacked = shouldUnpack if shouldUnpack
     node.files = {}
 
-  insertFile: (p, shouldUnpack, stat) ->
+  insertFile: (p, shouldUnpack, file, options, callback) ->
     dirNode = @searchNodeFromPath path.dirname(p)
     node = @searchNodeFromPath p
     if shouldUnpack or dirNode.unpacked
-      node.size = stat.size
+      node.size = file.stat.size
       node.unpacked = true
+      callback()
       return
 
-    # JavaScript can not precisely present integers >= UINT32_MAX.
-    if stat.size > 4294967295
-      throw new Error("#{p}: file size can not be larger than 4.2GB")
+    self = this
```

Comment:
You can use the `=>` function in CoffeeScript, which handles this automatically.